toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/MainApp.java

```diff
@@ -11,246 +11,236 @@
-							trayIcon.setImage(online_image);
-							connectedToWebApp = true;
-							NotificationManager.showMessage("Web Agent - Connected to Webapp !").showNotification();
-						}
-						else
-							NotificationManager.showMessage("The ApiKey does not match with the WebApp: \n" + webConfigProvider.get().getApiKey()).showNotification();
-					}
-					else 
-						NotificationManager.showMessage("The Web App does not anwser").showNotification();
-				} catch (IOException e1) {
-					LOG.error(e1.getMessage(), e1);
-				}
-	        }
-	    };
-	    return listener;
+	private PopupMenu initMenuItem() {
+		PopupMenu popup = new PopupMenu();
+	    
+	    MenuItem quitItem = new MenuItem("Quit");
+	    MenuItem connectItem = new MenuItem("Connect");
+	    MenuItem executeItem = new MenuItem("Execute Scripts");
+	    MenuItem startRecordingItem = new MenuItem("Start Recording");
+	    MenuItem stopRecordingItem = new MenuItem("Stop Recording");
+	    MenuItem settingsItem = new MenuItem("Settings");
+	    
+	    quitItem.addActionListener(this::killListener);
+	    connectItem.addActionListener(this::connectListener);
+	    executeItem.addActionListener(this::executeListener);
+	    stopRecordingItem.addActionListener(this::stopListener);
+	    startRecordingItem.addActionListener(this::start);
+	    settingsItem.addActionListener(this::settingsListener);
+	    
+	    popup.add(connectItem);
+	    popup.addSeparator();
+	    popup.add(startRecordingItem);
+	    popup.add(stopRecordingItem);
+	    popup.addSeparator();
+	    popup.add(executeItem);
+	    popup.addSeparator();
+	    popup.add(settingsItem);
+	    popup.addSeparator();
+	    popup.add(quitItem); 
+	    
+	    return popup;
 	}
 	
-	ActionListener getKillListener(){
-	    ActionListener listener = new ActionListener() {
-	        public void actionPerformed(ActionEvent e) {
-	        	agentServer.unRegister();
-	        	System.exit(-1);
-	        }
-	    };
-	    return listener;
+	private void connectListener(ActionEvent e){
+		connect();
 	}
 	
-	ActionListener getStartListener(){
-	    ActionListener listener = new ActionListener() {
-	        public void actionPerformed(ActionEvent e) {
-	        	try {
-	        		if(connectedToWebApp) {
-	        			boolean flag = false;
-		        		
-						if(verificationWebApp(chromeDriverName)) {
-							if(verificationWebApp(recorderName)) {
-								if(verificationWebApp(webAppName)) {
-									flag = true;
-									listenerStarted = true;
-									browserManager.startRecording();
-								}
-							}
-						}
-						
-						if(!flag) {
-							NotificationManager.showMessage("Unable to start recorder, please check recoder parameters !").showNotification();
-						}
-	        		}
-	        		else 
-						NotificationManager.showMessage("You have to be connected to the WebApp !").showNotification();
-				} catch (IOException e1) {
-					LOG.error(e1.getMessage(), e1);
+	//TODO: Switch implementation to websocket and have a connection watcher
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)